src/iomei.cpp

```diff
@@ -2893,26 +2893,40 @@ bool MEIInput::ReadDoc(pugi::xml_node root)
         return false;
     }
 
-    current = root.child("meiHead");
-    if (current.empty()) {
-        LogWarning("No header found in the MEI data, trying to proceed...");
+    if (root.attribute("meiversion")) {
+        std::string version = std::string(root.attribute("meiversion").value());
+        if (version == "5.0.0-dev") {
+            m_version = MEI_5_0_0_dev;
+        }
+        else if (version == "4.0.1") {
+            m_version = MEI_4_0_1;
+        }
+        else if (version == "4.0.0") {
+            m_version = MEI_4_0_0;
+        }
+        else if (version == "3.0.0") {
+            m_version = MEI_3_0_0;
+        }
+        else if (version == "2013") {
+            m_version = MEI_2013;
+        }
```

Comment:
Should this handle the case where it doesn't match any of these values?

Reply:
I guess not. Haven't thought about it though, as I just separated the version check from the header handling.

Reply:
By default setting `m_version` is `MEI_UNDEFINED`. I think it makes sense to go with that if we find a nonsense value in `meiversion`.